calicoctl/calicoctl.go

```diff
@@ -79,10 +81,11 @@ Options:
 		default:
 			fmt.Println(usage)
 		}
-	}
 
-	if err != nil {
-		fmt.Printf("Error executing command: %s\n", err)
-		os.Exit(1)
+		if err != nil {
+			fmt.Printf("Error executing command. Invalid option: 'calicoctl %s'. Use flag '--help' to read about a specific subcommand.\n", strings.Join(args, " "))
```

Comment:
In that case we shouldn't need to print any messages here (or even pass the errors up: that implies that action might be needed, whereas a contract of "succeeds, or os.exit(1)s" is nice and simple for all subcommands.